python/cudf/cudf/core/reshape.py

```diff
@@ -282,39 +315,40 @@ def concat(objs, axis=0, join="outer", ignore_index=False, sort=None):
         else:
             if axis == 0:
                 result = obj.copy()
+                if keys is not None:
+                    raise NotImplementedError(
+                        "Concatenation along axis = 0 "
+                        "when passing a dictionary is not supported yet."
+                    )
             else:
+                o_typ = typs.pop()
+                if o_typ not in allowed_typs:
+                    raise TypeError(
+                        f"cannot concatenate object of type {o_typ}"
+                    )
                 data = obj._data.copy(deep=True)
                 if isinstance(obj, cudf.Series) and obj.name is None:
                     # If the Series has no name, pandas renames it to 0.
                     data[0] = data.pop(None)
                 result = cudf.DataFrame._from_data(
                     data, index=obj.index.copy(deep=True)
                 )
+                if keys is not None:
+                    if isinstance(result, cudf.DataFrame):
```

Comment:
*question*: Does this do the right thing for the case where one is concatenating a dictionary of series?

```
pd.concat({"A": pd.Series([1, 2]), "B": pd.Series([3, 4])}, axis=1)
   A  B
0  1  3
1  2  4
```

Reply:
it does! see the test case here: https://github.com/rapidsai/cudf/pull/15623/files#diff-1d997d95893af6a665d5803d179f472c673a7d4e1a0a04a305bd2f1c4a66d957R1896